src/odemis/driver/technolution.py

```diff
@@ -279,6 +287,100 @@ def checkMegaFieldExists(self, mega_field_id, storage_dir):
                                        (mega_field_id, storage_dir), 200, raw_response=True)
         return json.loads(response.content)["exists"]
 
+    def getSamServiceVersion(self):
+        """
+        Get the SAM (Scanning and Acquisition Module) service software version via the ASM API.
+        :return (str): Version string of the SAM service software.
+        """
+        item_name = "sam_service_version"
+        response = self.asmApiGetCall("/monitor/item?item_name=" + item_name, 200, raw_response=True)
+        return response.text
+
+    # TODO Not yet implemented on simulator side, uncomment when implemented.
+    # def getSamRootfsVersion(self):
+    #     """
+    #     Makes monitor call to the ASM to retrieve the SAM root file system version.
+    #     :return (str):  Version string of the root file system of the SAM
+    #     """
+    #     item_name = "sam_rootfs_version"
+    #     response = self.asmApiGetCall("/monitor/item?item_name=" + item_name, 200, raw_response=True)
+    #     return response.text
+
+    def getSamFirmwareVersion(self):
+        """
+        Makes monitor call to the ASM to retrieve the SAM firmware version (Scanning and Acquisition Module).
+        :return (str):  Version string of the SAM firmware.
+        """
+        item_name = "sam_firmware_version"
+        response = self.asmApiGetCall("/monitor/item?item_name=" + item_name, 200, raw_response=True)
+        return response.text
+
+    def getAsmServiceVersion(self):
+        """
+        Makes monitor call to the ASM to retrieve the ASM service software version.
+        :return (str): Version string of the SAM service software.
+        """
+        item_name = "asm_service_version"
+        response = self.asmApiGetCall("/monitor/item?item_name=" + item_name, 200, raw_response=True)
+        return response.text
+
+    def getStateInstallation(self):
+        """
+        Checks the current installation status and returns the status of the last installation or if an installation is
+        in progress.
+        :return (str): one of the following values:
```

Comment:
The return doc does not match with your return values. The mapping with the dict is missing.